app/views/devise/two_factor_authentication/show.html.erb

```diff
@@ -7,5 +11,9 @@
   <%= submit_tag "Submit" %>
 <% end %>
 
-<%= link_to 'Resend Code', resend_code_user_two_factor_authentication_path, action: :get %>
-<%= link_to "Sign out", destroy_user_session_path, :method => :delete %>
\ No newline at end of file
+<% if current_user.direct_otp %>
+<%= link_to "Resend Code", resend_code_user_two_factor_authentication_path, action: :get %>
+<% else %>
+<%= link_to "Send me a code", resend_code_user_two_factor_authentication_path, action: :get %>
+<% end %>
```

Comment:
Another reason why this is flawed is because this would result in `send_new_otp` to be called in the TOTP scenario, and `send_new_otp` calls `create_direct_otp`, which doesn't apply to the TOTP scenario.